### PR TITLE
Lua5.3 grammar

### PR DESCRIPTION
**Pull request by ltratt**
Adds a Lua 5.3 grammar. Might be worth @fsfod having a quick once over of this before @ptersilie considers merging?

### CODE REVIEW

File: lua5_3/lua5_3.l [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+\[ "["
+\] "]"
+:: "::"
+; ";"
+: ":"
+, ","
+("(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?")|('(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?') "SHORT_STR"
+(\[\[.*?\]\])|(\[=\[.*?\]=\])|(\[==\[.*?\]==\]) "LONG_STR"
+(0[xX][0-9a-fA-F]+(\.[0-9a-fA-F]+)?([pP]-?[0-9a-fA-F]+)?)|(\.[0-9]+([eE]-?[0-9]+)?)|([0-9]+(\.[0-9]*)?([eE]-?[0-9]+)?) "NUMERAL"
+(--\[=\[.*?\]=\])|(--\[\[.*?\]\])|(--.*?$) ;
```
Comment (fsfod) [inline lua5_3/lua5_3.l]: If I'm reading this right this is the line that is trying to parse comments. What i was talking about was the number of equals between the between the opening brackets  can be more than one like this
``` Lua
--[==[]=]true]==]a = b
```
and it has to be ended with closing brackets that have the same number of equals.

Comment (ltratt) [reply lua5_3/lua5_3.l]: Correct on all fronts. Unfortunately, with normal (non-Perl) regular expressions we can't express "match '='s at the front and end". However, what I can do (which I've done in the most recent commit) is allow up to $n$ '='s. In this case I've allowed 5, which I suspect is more than anyone ever uses in practise. Do you think that's enough (or maybe even too many!)?

**Comment by ptersilie** [comment]
Do you want to keep the last commit or rebase it?

**Review by ptersilie — APPROVED** [review]

File: lua5_3/lua5_3.l (hunk 1 above)
Comment (fsfod) [reply lua5_3/lua5_3.l]: i don't think you would normally need to go over two most the time.

Comment (ltratt) [reply lua5_3/lua5_3.l]: OK, then I think 5 is plenty :)

**Comment by ltratt** [comment]
I've added a quick bit of text to the README about the lexer's limitation. If you're OK with that, I can squash.

File: lua5_3/README [hunk 2]
```diff
@@ -1,6 +1,6 @@
-============================================================================
+===============================================================================
   Lua 5.3 yacc grammar
-============================================================================
+===============================================================================
```
Comment (ptersilie) [inline lua5_3/README]: Why are these only 79?

Comment (ltratt) [reply lua5_3/README]: I can't count higher than 10 reliably. Fixed!

**Comment by ptersilie** [comment]
Looking good. Ready for squash!

**Comment by ltratt** [comment]
Squashed.

**Comment by ptersilie** [comment]
Merged!

**Pull request merged**